### PR TITLE
kam service add creates a new config directory.

### PR DESCRIPTION
**Pull request by gaganhegde**
**What type of PR is this?**
/kind bug

**What does this PR do / why we need it**:
The `kam service add` command does not create the `config` folder in the base directory. This fix corrects that error.

**Have you updated the necessary documentation?**

* [x] Documentation update is required by this PR.
* [x] Documentation has been updated.

**Which issue(s) this PR fixes**:
https://issues.redhat.com/browse/GITOPS-360

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: pkg/pipelines/environment_test.go [hunk 1]
```diff
@@ -237,6 +237,20 @@ func assertFileExists(t *testing.T, testFs afero.Fs, path string) {
 	}
 }
 
+func assertDirExists(t *testing.T, testFs afero.Fs, path string) {
+	t.Helper()
+	exists, err := afero.Exists(testFs, path)
```
Comment (bigkevmcd) [inline pkg/pipelines/environment_test.go:242]: You don't need to test for `Exists` and `DirExists` as they both fail if the path doesn't exist.

File: pkg/pipelines/service.go [hunk 2]
```diff
@@ -195,3 +200,19 @@ func createSvcImageBinding(cfg *config.PipelinesConfig, env *config.Environment,
 	resourceFilePath := makeImageBindingPath(cfg, filename)
 	return name, filename, res.Resources{resourceFilePath: triggers.CreateImageRepoBinding(cfg.Name, name, imageRepo, strconv.FormatBool(isTLSVerify))}
 }
+
+func createConfigFolder(m *config.Manifest, appFs afero.Fs, o *AddServiceOptions) error {
+	basePath, err := homedir.Expand(o.PipelinesFolderPath)
+	if err != nil {
+		return err
```
Comment (bigkevmcd) [inline pkg/pipelines/service.go:207]: how about expanding on this error a bit?

Getting the error [`cannot expand user-specific home dir`](https://github.com/mitchellh/go-homedir/blob/master/homedir.go#L68) probably isn't that useful to a user?

**Review by bigkevmcd — CHANGES_REQUESTED** [review]
I wonder if we shouldn't simplify `pkg/pipelines/ioutils/file_utils.go` to something like:

```go
func NewFilesystem() afero.Afero {
       return afero.Afero{Fs: afero.NewOsFs()}
}
```
Which would arguably improve this code to something like:

```go
isDir, err := testFs.DirExists(path)
```
What do you think?

**Comment by wtam2018** [comment]
Please update the doc.  Remove the node about create `config` folder if it does not exist.

**Review by bigkevmcd — APPROVED** [review]

**Comment by wtam2018** [comment]
/hold
@gaganhegde  please update docs.

File: pkg/pipelines/service.go [hunk 3]
```diff
@@ -195,3 +200,19 @@ func createSvcImageBinding(cfg *config.PipelinesConfig, env *config.Environment,
...
+	basePath, err := homedir.Expand(o.PipelinesFolderPath)
+	if err != nil {
+		return fmt.Errorf("Cannot expand the pipelines.yaml path : %s", o.PipelinesFolderPath)
+	}
+	env := m.GetEnvironment(o.EnvName)
+	app := m.GetApplication(o.EnvName, o.AppName)
+	servicePath := config.PathForService(app, env, o.ServiceName)
+	finalPath := filepath.Join(basePath, servicePath, "base", "config")
+	err = appFs.MkdirAll(finalPath, 0755)
+	if err != nil {
```
Comment (wtam2018) [inline pkg/pipelines/service.go:214]: Will this fail if directory already exists?

Comment (gaganhegde) [reply pkg/pipelines/service.go:214]: No William, it wont. if the directory is not present it will create one.

Comment (bigkevmcd) [reply pkg/pipelines/service.go:214]: https://golang.org/pkg/os/#MkdirAll

**Pull request merged**